NEXT-12782 + NEXT-12783 - Administration filter (#119)
shopware/docs: guides/plugins/plugins/administration/add-filter.md

# Add filter

## Overview

In this guide you'll learn, how to create a filter for the Shopware administration. A filter is just a little helper for formatting text.

> In this example, we create a filter that makes text into caps and adds an underscore at the beginning and end.

Reviewer: ```suggestion
In this example, we create a filter that converts text into uppercase and adds an underscore at the beginning and end.
```

## Prerequisites

This guide requires you to already have a basic plugin running.
If you don't know how to do this in the first place, have a look at our [Plugin base guide](../plugin-base-guide.md).

## Creating the filter

First we create a new file in the directory `<plugin root>/src/Resources/app/administration/src/app/filter`. 
In this case we name our filter `example`, so our file will be named `example.filter.js`.

Here's an example how your filter could look like:

{% code title="<plugin root>/src/Resources/app/administration/src/app/filter/example.filter.js" %}
```javascript
const { Filter } = Shopware;

Filter.register('example', (value) => {
    if (!value) {
        return '';
    }

    return `_${value.toLocaleUpperCase()}_`;
});
```
{% endcode %}

> As you can see, it's very simple. We use `Filter` from the `Shopware` object where we can register our filter with the function `register`.

Reviewer: ```suggestion
As you can see, it's very simple. We use `Filter` from the `Shopware` object where we can register our filter with the method `register`.
```

The first argument we pass is the name of our filter, which is `example`. The second argument is a function with which we format our text.

If you want to use multiple arguments in your filter function, it could look like this:

```javascript
Filter.register('example', (value, secondValue, thirdValue) => {
    ...
});
```

> Last, import the filter into your `main.js` file.

Reviewer: ```suggestion
Last, import the filter into your plugin's `main.js` file.
```
